Code/Framework/AzToolsFramework/AzToolsFramework/API/ComponentModeCollectionInterface.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/Outcome/Outcome.h>
+#include <AzToolsFramework/API/ViewportEditorModeTrackerNotificationBus.h>
+
+namespace AzToolsFramework
+{
+    //! The AZ::Interface for component mode collection queries.
+    class ComponentModeCollectionInterface
+    {
+    public:
+        AZ_RTTI(ComponentModeCollectionInterface, "{DFAA4450-BBCD-47C0-9B91-FEA2DBD9B152}");
+
+        virtual ~ComponentModeCollectionInterface() = default;
+
+        //! Retrieves the list of all Component types (usually one).
+        virtual const AZStd::vector<AZ::Uuid>& GetComponentTypes() const = 0;
```

Comment:
It might also be worth commenting a precondition of the function is to call it when _in_ a Component Mode (otherwise the list will be empty and meaningless?)

Reply:
good idea. this call now returns a copy of the aforementioned data when in component mode, otherwise empty vec 👍